libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -628,13 +655,72 @@ void CodeTransform::visitExpression(Expression const& _expression)
 	expectDeposit(1, height);
 }
 
+void CodeTransform::setupReturnVariablesAndFunctionExit()
+{
+	yulAssert(!m_functionExitStackHeight.has_value(), "");
+	yulAssert(m_scope, "");
+
+	ScopeGuard scopeGuard([oldScope = m_scope, this] { m_scope = oldScope; });
+	if (!m_scope->functionScope)
```

Comment:
Can this function be called on the outermost block (i.e. the one that is not inside a function)?

Reply:
If this function is called directly from the ``FunctionDefinition`` visit, ``m_scope`` will be the virtual function scope. If allocation is deferred, this will only be called later during the function body, in which case the parent scope should be the virtual function scope.

Reply:
Is it possible to add assertions somehow? We never know from where this function will be called in the future.

Reply:
I could instead set ``subTransform.m_scope`` to the function body scope and not the virtual function scope in the function definition visit... then I could assert ``m_scope->superScope && m_scope->superScope->functionScope`` here...

Currently it basically asserts ``m_scope->functionScope || (m_scope->superScope && m_scope->superScope->functionScope)``, so it's not that much of a difference...